source/grid/grid_tools.cc

```diff
@@ -1060,7 +1061,18 @@ namespace GridTools
     const Triangulation<dim, spacedim> &tria = mesh.get_triangulation();
 
     const std::vector< Point<spacedim> > &vertices = tria.get_vertices();
-    const std::vector< bool       > &used     = tria.get_used_vertices();
+
+    Assert ( tria.get_vertices().size() == marked_vertices.size() || marked_vertices.size() ==0,
+             ExcDimensionMismatch(tria.get_vertices().size(), marked_vertices.size()));
+
```

Comment:
@vishalkenchan i would add another `Assert` here: 

`marked_vertices` should have `true` only in those places where `tria.get_used_vertices()` is true. In other words, it should be only allowed to provide a mask which is a subset of vertices on active cells. Please check if there is something in `std::` that can help with this, otherwise one could add to `.cc`:
```
namespace
{
   bool check_mask(const std::vector<bool> & marked_vertices, const std::vector<bool> & used_vertices)
   {
      // code
      return ...
   }
}
```
and then
```
Assert (check_mask(marked_vertices, tria.get_used_vertices(),
        ExcMessage("marked_vertices should be a subset of used vertices in the triangulation"));
```
Since this is done in `Debug` only, we don't really care that it takes extra time.